nuxeo-tree-snapshot-core/src/main/java/org/nuxeo/snapshot/SnapshotableAdapter.java

```diff
@@ -294,11 +320,11 @@ protected DocumentModel restore(DocumentModel leafVersion, DocumentModel target,
 
         if (target.isFolder() && first) {
             // save all subtree
-            olddocs = session.query("select * from Document where ecm:path STARTSWITH '" + target.getPathAsString()
-                    + "'");
+            olddocs = session.query(
+                    "select * from Document where ecm:path STARTSWITH '" + target.getPathAsString() + "'");
```

Comment:
Sorry I missed this (not part of your changes but was reformatted): please do proper escaping of the path.